### PR TITLE
fix `ServiceWorker.register should reject a Promise for unsecure urls` (close #1411)

### LINKED ISSUE #1411: ServiceWorker.register should reject a Promise for unsecure urls
Reproduced on http://skanska-poc-react.azurewebsites.net/login

Secure origins:
```js
(https, *, *)
(wss, *, *)
(*, localhost, *)
(*, 127/8, *)
(*, ::1/128, *)
(file, *, —)
(chrome-extension, *, —) 
```

From https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/node/window.js [hunk 1]
```diff
@@ -304,8 +309,15 @@ export default class WindowSandbox extends SandboxBase {
 
         if (nativeMethods.registerServiceWorker) {
             window.navigator.serviceWorker.register = (...args) => {
-                if (typeof args[0] === 'string')
-                    args[0] = getProxyUrl(args[0], { resourceType: stringifyResourceType({ isScript: true }) });
+                const url       = args[0];
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: Move this code to the `_isSecureOrigin (url)` and 
add link https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features as comment inside function

File: src/client/sandbox/node/window.js [hunk 2]
```diff
@@ -29,6 +31,9 @@ const arrayConcat = Array.prototype.concat;
 
 const HTTP_PROTOCOL_RE = /^http/i;
 
+const ALLOWED_SERVICE_WORKER_PROTOCOLS  = ['https:', 'wss:', 'file:', 'chrome-extension:'];
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: `chrome-extention` is not necessary. Url with `chrome-extention` protocol will not modified by `getProxyUrl` function and browser will perform secure origin validation itself.

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
ping @AlexanderMoskovkin

**Pull request merged**